src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,51 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducerImpl bulldozerProducer;
+    private ExcavatorProducerImpl excavatorProducer;
+    private TruckProducerImpl truckProducer;
+
+    public MachineServiceImpl() {
+    }
+
+    public MachineServiceImpl(BulldozerProducerImpl bulldozerProducer,
+                              ExcavatorProducerImpl excavatorProducer,
+                              TruckProducerImpl truckProducer) {
+        this.bulldozerProducer = bulldozerProducer;
+        this.excavatorProducer = excavatorProducer;
+        this.truckProducer = truckProducer;
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machineList.addAll(bulldozerProducer.get());
+        } else if (type.equals(Excavator.class)) {
+            machineList.addAll(excavatorProducer.get());
+        } else {
```

Comment:
Don't return null or specific type List in `getAll()` when no producer has been found. It's better to return an empty List to avoid potential `NullPointerException`. Consider adding an `else if` check for the Truck type and a final `else` block that returns an empty list.